### PR TITLE
Fixed memory leak in the QUIC stream manager

### PR DESCRIPTION
**Pull request by bryancall**
Removed the proxy allocator for the streams.  They can be added back in if there is a performance issue.

### CODE REVIEW

File: iocore/net/quic/QUICStreamManager.h [hunk 1]
```diff
@@ -40,6 +40,12 @@ class QUICStreamManager : public QUICFrameHandler, public QUICFrameGenerator
 {
 public:
   QUICStreamManager(QUICContext *context, QUICApplicationMap *app_map);
+  ~QUICStreamManager()
+  {
+    for (auto stream = stream_list.pop(); stream != nullptr; stream = stream_list.pop()) {
+      _stream_factory.delete_stream(stream);
+    }
+  }
```
Comment (bneradt) [inline iocore/net/quic/QUICStreamManager.h:48]: Probably best to declare `~QUICStreamManager()` here and define in the `.cc` file. That will be consistent with the rest of the functions here.

**Review by bneradt — CHANGES_REQUESTED** [review]
Looks good.

**Comment by bryancall** [comment]
[approve ci autest]

**Review by bneradt — APPROVED** [review]

**Comment by bryancall** [comment]
[approve ci ubuntu]

**Comment by bryancall** [comment]
[approve ci autest]

**Review by maskit — APPROVED** [review]

**Comment by maskit** [comment]
[approve ci autest]

**Comment by bryancall** [comment]
[approve ci autest]

**Comment by maskit** [comment]
[approve ci autest]

**Comment by bneradt** [comment]
[approve ci autest]

**Pull request merged**